clang-tools-extra/test/lit.cfg.py

```diff
@@ -69,8 +70,15 @@
 clang_tidy_headers = os.path.join(
     config.test_source_root, "clang-tidy", "checkers", "Inputs", "Headers"
 )
+
 config.substitutions.append(("%clang_tidy_headers", clang_tidy_headers))
 
+clang_doc_install = os.path.join(
+    os.path.dirname(shutil.which("clang-doc")), "..", "share", "clang"
+)
```

Comment:
I don't think we wan't to be using something equivalent to `which clang-doc` in the lit config. First, I don't see this pattern used anywhere else in LLVM. Second, its completely unclear to me why this is necessary.

Reply:
I'd wanted a way to copy index.js to the install directory is there another way to accomplish that?

Reply:
what do you mean by "install directory"? the directory that we install toolchain components into w/ `ninja install`? the place where you're running tests? or something else?

Reply:
when the windows runner is building it isn't copying the asset files to the ${CMAKE_INSTALL_DATADIR}/clang defined here https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-doc/tool/CMakeLists.txt
you can view some the logs here
https://buildkite.com/llvm-project/github-pull-requests/builds/68626#018fcd51-66a7-4091-b861-7f756301a191

I figured i would just manually copy the files in run directives for the test cases

Reply:
so the CMAKE_INSTALL_DATADIR variable is used when you build the install target w/ `ninja install` or equivalent. The install target isn't the build of clang/llvm or even clang-doc. Instead it copies the artifacts you want to install into their final location. Essentially, imagine if you wanted to install clang to`/usr/bin/`, and all its headers to `/usr/include/`, and all its libs to `/usr/lib/`. That's basically what the install step is for, and you can point the base install directory to be anywhere. For instance I have a `$HOME/clang-install/` folder that I install the toolchain into when I need to use it in production type situations. For daily development, I just run things out of my `build` folder. The build folder is also where tests run.

This may be hidden by your IDE, but esentialy, you point CMake at the project, and tell it to build everything in some folder, usually called `build`.  After CMake runs and sets up all the build files for a build system, like `ninja` or `make`,  you can build the project by using the right build command. For clang-doc, that would be something like `ninja clang-doc` or `ninja check-clang-extra-clang-doc` to run its tests. But that won't install anything, it just builds things in the `build` directory. To install you'd run `ninja install` and then only the things that were important to ship would be installed to the location you specified, or to `/usr/local/` on Linux if you didn't specify anything.

The default style sheet (and any other fixed resources) should be easy for you to either copy to the right place, or to pass into the `clang-doc` command line, since its path relative to the source directory is fixed. @petrhosek may know of some other mechanism to force an asset to be copied into the build directory, but I'm not sure that is required in this case.

Reply:
I will note that currently there isn't a mechanism to pass another default stylesheet to the clang-doc the option               --stylesheets=<string> in clang-doc only extends the css stylesheet but still requires default stylesheet to be present.
I was trying to find a ways to copy the asset via the lit command line but i can't seem to do it

Reply:
actually that's probably a problem with clang-doc default behaviour. It should be that --stylesheet should replace the default stylesheet not extend it

Reply:
That's also likely true, but TBH I don't have a problem running clang-doc from the build directory, so maybe something else is wrong?